campaigns.py

```diff
@@ -6,52 +6,87 @@
 
 from jinja2 import Template
 
-import vm
+import app
+
+
+env['eagerly_disconnect'] = True
+
+
+APPLICATIONS = ['frontend', 'static']
+CAMPAIGN_CLASSES = ['red', 'black', 'green']
+
 
 def validate_classes(campaign_class):
     """Checks that the campaign class is valid"""
-    if campaign_class in ['red', 'black', 'green']:
+    if campaign_class in CAMPAIGN_CLASSES:
         return campaign_class
-    raise Exception, "Invalid class %s, valid values are 'red', 'black', 'green'" % campaign_class
+    raise Exception, "Invalid class {}, valid values are {}".format(campaign_class, CAMPAIGN_CLASSES)
 
 
 @runs_once
-def homepage_template():
-    """Promotes a campaign to the homepage of GOV.UK"""
-    context = {
-        'heading': prompt("Heading for campaign:"),
-        'extra_info': prompt("Extra information for campaign:"),
-        'more_info': prompt("Link for more information:"),
-        'campaign_class': prompt("Campaign class:", validate=validate_classes)
+def set_context():
+    env['context'] = {
+        'heading': prompt("Heading for campaign:", 'heading'),
+        'extra_info': prompt("Extra information for campaign:", 'extra_info'),
+        'more_info': prompt("Link for more information:", 'more_info'),
+        'campaign_class': prompt("Campaign class:", 'campaign_class', validate=validate_classes)
     }
 
-    template = Template("""<div id="campaign" class="{{ campaign_class }}">
-  <div class="campaign-inner">
-    <h1>{{ heading|e }}</h1>
-    <p>{{ extra_info|e }}</p>
-    <a href="{{ more_info|e }}">More information</a>
-  </div>
-</div>""")
-    env['template_contents'] = template.render(context)
 
-    print "Template contents:\n%s" % env['template_contents']
+def template(app):
+    if app == 'frontend':
+      template = Template("""<div id="campaign" class="{{ campaign_class }}">
+    <div class="campaign-inner">
+      <h1>{{ heading|e }}</h1>
+      <p>{{ extra_info|e }}</p>
+      <a href="{{ more_info|e }}">More information</a>
+    </div>
+  </div>""")
+    elif app == 'static':
+      template = Template("""<p>{{ heading|e }}<br />
+    {{ extra_info|e }}</p>
+  <a href="#" class="right">{{ more_info|e }}</a>""")
+
+    env['template_contents'] = template.render(env.context)
+
 
 @task
 @roles('class-frontend')
-def deploy_to_homepage():
-    execute(homepage_template)
+def deploy_banner(application):
+    execute(template, application)
+    if application == 'frontend':
+        remote_filename = '/var/apps/frontend/app/views/root/_campaign_notification.html.erb'
+    elif application == 'static':
+        remote_filename = "/var/apps/static/app/views/notifications/banner_{}.erb".format(env.campaign_class)
+    content = env['template_contents']
+    put(StringIO.StringIO(content), remote_filename, use_sudo=True, mirror_local_mode=True)
+    sudo('chown deploy:deploy %s' % remote_filename)
```

Comment:
Yes. When using `put` the file is chown'ed by the person running the fabric task.

```
joshmyers@migration-staging-frontend-1:~$ ls -al /var/apps/frontend/app/views/root/_campaign_notification.html.erb
-rw-r--r-- 1 deploy deploy 168 Aug 25 08:53 /var/apps/frontend/app/views/root/_campaign_notification.html.erb
```
```
In [1]: put(StringIO.StringIO(content), remote_filename, use_sudo=True, mirror_local_mode=True)
[frontend-1.frontend] put: <file obj> -> /var/apps/frontend/app/views/root/_campaign_notification.html.erb
Out[1]: ['/var/apps/frontend/app/views/root/_campaign_notification.html.erb']
```
```
joshmyers@migration-staging-frontend-1:~$ ls -al /var/apps/frontend/app/views/root/_campaign_notification.html.erb
-rw-r--r-- 1 joshmyers joshmyers 168 Aug 25 08:54 /var/apps/frontend/app/views/root/_campaign_notification.html.erb
```